cpp/include/cugraph/prims/reduce_v.cuh

```diff
@@ -54,7 +54,7 @@ T reduce_v(raft::handle_t const& handle,
   auto ret = thrust::reduce(rmm::exec_policy(handle.get_stream())->on(handle.get_stream()),
                             vertex_value_input_first,
                             vertex_value_input_first + graph_view.get_number_of_local_vertices(),
-                            init);
+                            (handle.get_comms().get_rank() == 0) ? init : T{0});
```

Comment:
This code will seg-fault in SG. No comms in SG. T{0} will not work if T is a thrust::tuple.

You may do something like

```
(GraphView::is_multi_gpu && handle.get_comms().get_rank() == 0) ? init : T{}
```
Instead.

Reply:
And reduce_v should work for tuple of arithmetic types as well, so I think we'd better add a test case with thrust::zip_iterator (of arithmetic types).

Reply:
I am not sure that reduce_v works with a zip iterator.
This is what I get 
```
thrust-src/thrust/functional.h(234): error: no operator "+" matches these operands
            operand types are: const thrust::tuple<int, int, float, thrust::null_type, thrust::null_type, thrust::null_type, thrust::null_type, thrust::null_type, thrust::null_type, thrust::null_type> + const thrust::tuple<int, int, float, thrust::null_type, thrust::null_type, thrust::null_type, thrust::null_type, thrust::null_type, thrust::null_type, thrust::null_type>
          detected during:
            instantiation of "T thrust::plus<T>::operator()(const T &, const T &) const [with T=thrust::tuple<int, int, float, thrust::null_type, thrust::null_type, thrust::null_type, thrust::null_type, thrust::null_type, thrust::null_type, thrust::null_type>]" 
```
My understanding is that the call to thrust::reduce inside reduce_v implicitly tries to instantiate the thrust::plus<thrust::tuple<...>> which does not make sense because operator+ is not defined for thrust::tuple as far as I am aware.

Reply:
Oh, yes, you're right. Let's submit a cuGraph issue to fix this.

Reply:
Or if you want to fix this in this PR, you can refer to 

https://github.com/rapidsai/cugraph/blob/branch-21.08/cpp/include/cugraph/prims/copy_v_transform_reduce_in_out_nbr.cuh#L127

```plus_edge_op_result``` should work with thrust::tuple.